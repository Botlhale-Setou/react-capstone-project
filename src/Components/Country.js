import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const Country = (props) => {
  const { name, flag, area } = props;

  return (
    <Link to={`/details/${name}`} className="countryCard">
      <h2 className="countryName">
        {name}
        <span>
          {area}
          km&#178;
        </span>
      </h2>
      <img src={flag} alt={`${name} flag`} className="countryFlag" />
    </Link>
  );
};

Country.propTypes = {
  name: PropTypes.string,
  flag: PropTypes.string,
  area: PropTypes.number,
}.isRequired;

export default Country;
