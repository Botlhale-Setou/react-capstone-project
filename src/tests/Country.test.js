import React from "react";
import Renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import store from "../Redux/ConfigureStore";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import Country from "../Components/Country";

describe("Country test", () => {
  test("Country render", () => {
    const tree = Renderer.create(
      <Provider store={store}>
        <Router>
          <Country />
        </Router>
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
