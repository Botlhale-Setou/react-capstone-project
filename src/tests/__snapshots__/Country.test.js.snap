// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country test Country render 1`] = `
<a
  className="countryCard"
  href="/details/undefined"
  onClick={[Function]}
>
  <h2
    className="countryName"
  >
    <span>
       kmÂ²
    </span>
  </h2>
  <img
    alt="undefined flag"
    className="countryFlag"
  />
</a>
`;
